### ISSUE TITLE
Source menu should be ordered somehow

### ISSUE THREAD

**Issue by annamorphism** [DDMAL/CantusDB#1536]
The edit-sequence page allows the assignment of chants to a source in the database, but the dropdown is not in any kind of order, so it's hard to find the right one:
![image](https://github.com/DDMAL/CantusDB/assets/67451875/879e33d7-879a-48b5-9a4f-b78468d01769)

We don't really expect to have to do this after the Bower data is cleaned up, so it's not a priority, but it would make the cleaning up easier...

**Comment by dchiller** [comment]
Agreed. I think this is something we will get to in the broader overhaul of segments and projects ... I'm not sure exactly how, but it will definitely be different/better than this.

**Comment by annamorphism** [comment]
wait, they are ordered, just by the city, with special characters at the end of the alphabet (which happened to be where I looked first). So not quite as bad as I thought...
<img width="490" alt="image" src="https://github.com/DDMAL/CantusDB/assets/67451875/b801885e-e79a-4fdc-ac83-9bb7f7ba9993">

**Comment by ahankinson** [comment]
It’s quite obviously because Unicode code points for non-Latin characters are outside of the range 0-127 and thus their numeric value is higher, and OBVIOUSLY when sorting text you want to sort by the numerical code point and not by their lexical value.

Says every database ever.

This is computing law that was written in the 1960s when every character of storage cost the same as a new car, and must never be changed

(Yes, it’s dumb)

**Comment by annamorphism** [comment]
> (Yes, it’s dumb)

speaking from personal experience with lowercase "d" falling after uppercase Z...agreed.
